protos/protoc.sh

```diff
@@ -1,9 +1,33 @@
 #!/usr/bin/env bash
-
+#
+# protoc.sh uses the local protoc if installed, otherwise
+# docker will be used with a complete environment provided
+# by https://github.com/jaegertracing/docker-protobuf.
+# Alternative images like grpc/go are very dated and do not
+# include the needed plugins and includes.
+#
 set -e
 
+GOPATH=${GOPATH:-$(go env GOPATH)}
+
+if command -v protoc >/dev/null; then
+	GW_PATH="$GOPATH"/src/github.com/grpc-ecosystem/grpc-gateway
+	GO111MODULES=on go get github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway
```

Comment:
Fetching should be unnecessary since its a required dep and should have already been fetch by `go` at the correct version. That being the case you can also get the file from the go mod src path, though this requires knowing the version number used. For example I've got:
```sh
ls -1d /home/manny/go/pkg/mod/github.com/grpc-ecosystem/grpc-gateway@*/
/home/manny/go/pkg/mod/github.com/grpc-ecosystem/grpc-gateway@v1.14.6/
/home/manny/go/pkg/mod/github.com/grpc-ecosystem/grpc-gateway@v1.14.7/
/home/manny/go/pkg/mod/github.com/grpc-ecosystem/grpc-gateway@v1.14.8/
```
We can pull out the version number from go.mod pretty easily though.

Reply:
We could build the container with the values that we want to use: https://github.com/jaegertracing/docker-protobuf/blob/master/Dockerfile#L1-L10.

A series or `go list -m all | grep github.com/grpc-ecosystem/grpc-gateway | awk '{print $2}'` (for each dependency) and `docker build --build-arg ...` could do this if we want to build our own version of the image.

Reply:
Yeah, and this PR doesn't make things worse in that respect anyway, so :+1: